### ISSUE TITLE
Normal Loss

### ISSUE THREAD

**Issue by rohun-tripathi** [JasonQSY/surface_normal#1]
Hi, I can't find the code for the normal loss in your training code. Are you using angle based surface normal loss?

Thanks

**Comment by JasonQSY** [comment]
I write the loss in `task/loss.py`. we purpose a novel loss function
<img width="485" alt="screen shot 2018-10-02 at 7 36 04 pm" src="https://user-images.githubusercontent.com/15360639/46382721-7ff6f180-c67a-11e8-9659-14277df506d6.png">
motivated by https://arxiv.org/abs/1706.07593

**Comment by rohun-tripathi** [comment]
Did you experiment with negative of inner product (-N.N') as they do in the paper you shared? Does that alter results considerably, when compared with results from training using arccos(N.N')?

**Comment by JasonQSY** [comment]
I think it depends on the dataset. For the dataset I trained on, arccos outperforms the inner product significantly, while the combined loss is the best.
